From 80d2c97a4e05695062806aed3839d78b35c3a91f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:35:56 -0400
Subject: [alsa] hda: Fix usage of "model" module parameter

Message-id: <140972823573-347-patchset-jkysela@redhat.com>
Patchwork-id: 91043
O-Subject: [RHEL 7.1 PATCH 346/442] ALSA: hda - Fix usage of "model" module parameter
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

A recent refactoring broke the possibility to manually specify
model name as a module parameter. This patch restores the desired
functionality.

Fixes: c21c8cf77f47 ('ALSA: hda - Add fixup_forced flag')
Reported-by: Kent Baxley <kent.baxley@canonical.com>
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8fffe7d1f094eea88aa0380255e247b285a2d5f2)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_auto_parser.c b/sound/pci/hda/hda_auto_parser.c
index b684c6e..dabe419 100644
--- a/sound/pci/hda/hda_auto_parser.c
+++ b/sound/pci/hda/hda_auto_parser.c
@@ -898,6 +898,7 @@ void snd_hda_pick_fixup(struct hda_codec *codec,
    if (!strcmp(codec->modelname, models->name)) {
     codec->fixup_id = models->id;
     codec->fixup_name = models->name;
+    codec->fixup_list = fixlist;
     codec->fixup_forced = 1;
     return;
    }
-- 
1.7.1